neurogym/wrappers/monitor.py

```diff
@@ -461,3 +496,120 @@ def plot_training_history(
             logger.info(f"Figure saved to {save_path}")
 
         return fig
+
+    def record_activations(
+        self,
+        layer: nn.Module,
+        name: str | None = None,
+        steps: int | None = None,
+    ) -> ActivationMonitor:
+        """Record the output activations of a layer over a trial.
+
+        Args:
+            layer: The layer whose activations are being monitored.
+            name: The name to use for the activation monitor.
+                This can be useful for retrieving activation monitors
+                at a later stage.
+            steps: The steps to record for.
+                This could be less than the total number of steps in a trial.
+
+        Returns:
+            An ActivationMonitor instance.
+        """
+        # Ensure that we have a name for this monitor
+        if name is None:
+            name = layer.__class__.__name__
+
+        # The total number of steps to record during each trial.
+        if steps is None:
+            steps = int(self.tmax / self.dt)
+
+        # Create an activation monitor
+        am = ActivationMonitor(layer, steps, name)
+
+        # Check if we are replacing an existing monitor
+        if name in self.activations:
+            logger.warning(f"Replacing an existing activation monitor '{name}'.")
+
+        # Store and return the monitor
+        self.activations[name] = am
+
+        return am
+
+    def plot_activations(
+        self,
+        name: str,
+        population: str,
+        figsize: tuple[int, ...] | None = None,
+        neurons: list | None = None,
+        mean: bool = False,
+    ) -> tuple[plt.Figure, plt.Axes]:
+        """Plot the neuron activations.
+
+        Args:
+            name: Name of the layer.
+            population: The neuron population to plot.
+            figsize: The size of the figure.
+            neurons: List of neurons to plot. If None, all neurons will be plotted.
+            mean: If set, plot the mean activation over all trials rather than each separate trial.
+
+        Raises:
+            ValueError: Raised if there is no such layer in the history.
+
+        Returns:
+            A plot of all neuron activations.
+        """
+        # First, get the right layer
+        am = self.activations.get(name)
+        if am is None:
+            msg = f"Invalid layer name '{name}'"
+            raise ValueError(msg)
+
+        # Get the activations for the specified population
+        activations = np.array(am.history.get(population))
+        if activations is None:
+            msg = f"Invalid population '{population}' (available populations: {am.monitor.populations})"
+            raise KeyError(msg)
+
+        # Neuron selection
+        if neurons is None:
+            neurons = list(range(activations[0].shape[1]))
+
+        if max(neurons) > activations[0].shape[1] - 1:
+            msg = "Some of the requested neuron IDs are beyond the size of the layer."
```

Comment:
Add extra info to the error, so the user knows what the limits are.

```suggestion
        total_neurons = activations[0].shape[1] - 1
        if max(neurons) > total_neurons:
            msg = "Some of the requested neuron IDs are beyond the size of the layer ({total_neurons })."
```